gamestonk_terminal/due_diligence/sec_view.py

```diff
@@ -0,0 +1,93 @@
+""" SEC View """
+__docformat__ = "numpy"
+
+import argparse
+from typing import List
+from datetime import datetime
+import requests
+import pandas as pd
+from bs4 import BeautifulSoup
+from gamestonk_terminal.helper_funcs import (
+    get_user_agent,
+    check_positive,
+    parse_known_args_and_warn,
+)
+
+
+def fails_to_deliver(other_args: List[str], ticker: str):
+    """Display fails-to-deliver data for a given ticker
+
+    Parameters
+    ----------
+    other_args : List[str]
+        argparse other args - ["-n", "10"]
+    ticker : str
+        Stock ticker
+    """
+
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="ftd",
+        description="""Prints latest fails-to-deliver data. [Source: SEC]""",
+    )
+    parser.add_argument(
+        "-n",
+        "--num",
+        action="store",
+        dest="n_num",
+        type=check_positive,
+        default=10,
+        help="number of latest fails-to-deliver being printed",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        url_ftds = "https://www.sec.gov/data/foiadocsfailsdatahtm"
+        text_soup_ftds = BeautifulSoup(
+            requests.get(url_ftds, headers={"User-Agent": get_user_agent()}).text,
+            "lxml",
+        )
+
+        table = text_soup_ftds.find("table", {"class": "list"})
+        links = table.findAll("a")
+        link_idx = 0
+        ftds_data = pd.DataFrame()
+        while len(ftds_data) < ns_parser.n_num:
+            if link_idx > len(links):
+                break
+            link = links[link_idx]
+            url = "https://www.sec.gov" + link["href"]
+            all_ftds = pd.read_csv(
+                url,
+                compression="zip",
+                sep="|",
+                engine="python",
+                skipfooter=2,
+                usecols=[0, 2, 3, 5],
+                dtype={"QUANTITY (FAILS)": "int"},
+            )
+            tmp_ftds = all_ftds[all_ftds["SYMBOL"] == ticker]
+            del tmp_ftds["PRICE"]
+            del tmp_ftds["SYMBOL"]
+            # merge the data from this archive
+            ftds_data = pd.concat([ftds_data, tmp_ftds], ignore_index=True)
+            link_idx += 1
+
+        # clip away extra rows
+        ftds_data = ftds_data.sort_values("SETTLEMENT DATE", ascending=False)[
+            0 : ns_parser.n_num
+        ]
+        ftds_data["SETTLEMENT DATE"] = ftds_data["SETTLEMENT DATE"].apply(
+            lambda x: datetime.strptime(str(x), "%Y%m%d").strftime("%Y-%m-%d")
+        )
+        print(
+            ftds_data.rename(
+                columns={"SETTLEMENT DATE": "Date", "QUANTITY (FAILS)": "Quantity"}
+            ).to_string(index=False)
+        )
```

Comment:
Missing a print("")

Reply:
should be on line 86 :)

Reply:
I meant an additional one, to give a new line between last print and new command.

<img width="169" alt="Captura de ecrã 2021-04-18, às 23 33 24" src="https://user-images.githubusercontent.com/25267873/115162978-7c8fae00-a09e-11eb-843d-aa79b1b80b4e.png">

Reply:
:facepalm:, gotcha!
Mmm, do you want to have console output and the bar plot?

Reply:
Nah, I don't think is needed. That dataframe is the one that your model should output, and the view digest :)